oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java

```diff
@@ -275,6 +278,34 @@ public static String encode(String s) {
         return '\"' + s + '\"';
     }
 
+    /**
+     * Similar to #encode(String) but appends the result to the given buffer. This method exists to avoid the allocation
+     * of intermediate objects to represent the encoded String when the caller already has a buffer where the result
+     * should be appended.
+     *
+     * @param s the text to convert
+     * @param buff the target buffer
+     * @return the target buffer with the encoded string appended
+     */
+    private static StringBuilder encode(String s, StringBuilder buff) {
+        if (s == null) {
+            return buff.append("null");
+        }
+        int length = s.length();
+        if (length == 0) {
+            return buff.append("\"\"");
```

Comment:
I personally find it easier to *read*, when all returns are on the same level of nesting (so this would require else/if etc). But this may be a matter of taste.

Reply:
As this method is a variation of `StringBuilder encode(String s)`, I kept it as close as possible to the other method with only the changes needed to use an existing `StringBuilder`, without doing any stylist changes, as this seems to be outside the scope of this PR.

Reply:
good point